Add console message tests for Web Inspector with various iframe conditions.

basuke opened the pull request:
#### 6665448ee42a0b6d34a214b5759dc567c257d803
<pre>
Add console message tests for Web Inspector with various iframe conditions.
<a href="https://bugs.webkit.org/show_bug.cgi?id=300654">https://bugs.webkit.org/show_bug.cgi?id=300654</a>
<a href="https://rdar.apple.com/162555044">rdar://162555044</a>

Reviewed by BJ Burg.

Add tests for console message that ensures messages are received by Web Inspector from:
- same/cross origin iframe.
- same origin grand child iframe embedded in same/cross origin iframe.

* LayoutTests/http/tests/inspector/console/message-from-iframe-expected.txt: Added.
* LayoutTests/http/tests/inspector/console/message-from-iframe.html: Added.
* LayoutTests/http/tests/inspector/console/resources/console-messages.html: Added.
* LayoutTests/http/tests/inspector/console/resources/embedded-cross-origin.html: Added.
* LayoutTests/http/tests/inspector/console/resources/embedded-same-origin.html: Added.

Canonical link: <a href="https://commits.webkit.org/301498@main">https://commits.webkit.org/301498@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5e01b4f47824e3e6aea6be1dd572b1dcdb921442

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126153 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45805 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36609 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/132955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77951 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/25b40538-4efb-425d-af2c-7a9903980cad) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46479 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54350 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/132955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64177 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0d2f889f-1d76-46f2-a2a4-6d57578ebadc) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129101 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37190 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112845 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/132955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36090 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31027 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/76429 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106973 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31250 "Passed tests") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/135663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52907 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40636 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/135663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109062 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/135663 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26579 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49693 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28031 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50303 "Hash 5e01b4f4 for PR 52264 does not build (failure)") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52804 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52132 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55477 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53842 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

the-chenergy commented on LayoutTests/http/tests/inspector/console/resources/embedded-cross-origin.html:
```diff
@@ -0,0 +1,11 @@
+iframe.onload = () => console.log("iframe is loaded in " + location.href);
+document.body.appendChild(iframe);
+
+</script>
```
Nit: add new line at end of file.

the-chenergy commented on LayoutTests/http/tests/inspector/console/resources/embedded-cross-origin.html:
```diff
@@ -0,0 +1,11 @@
+<script>
+
+const iframe = document.createElement("iframe");
+const crossOriginHost = location.hostname === 'localhost' ? '127.0.0.1' : 'localhost';
```
Nit: use double quotes.

burg commented on LayoutTests/http/tests/inspector/console/message-from-iframe.html:
```diff
@@ -0,0 +1,63 @@
+    function test() {
+        // The purpose of these tests are to test that Web Inspector should receive console messages generated from various iframe tree correctly.
+
+        const suite = InspectorTest.createAsyncSuite("message-from-iframe");
```
Style nit: most async suite names are CamelCase so like `MessageFromIFrame`, `CrossOriginIFrame`, etc.

basuke committed:
Add console message tests for Web Inspector with various iframe conditions.
https://bugs.webkit.org/show_bug.cgi?id=300654
rdar://162555044

Reviewed by BJ Burg.

Add tests for console message that ensures messages are received by Web Inspector from:
- same/cross origin iframe.
- same origin grand child iframe embedded in same/cross origin iframe.

* LayoutTests/http/tests/inspector/console/message-from-iframe-expected.txt: Added.
* LayoutTests/http/tests/inspector/console/message-from-iframe.html: Added.
* LayoutTests/http/tests/inspector/console/resources/console-messages.html: Added.
* LayoutTests/http/tests/inspector/console/resources/embedded-cross-origin.html: Added.
* LayoutTests/http/tests/inspector/console/resources/embedded-same-origin.html: Added.

Canonical link: https://commits.webkit.org/301498@main
